cpp/community_detection_module/community_detection_module.cpp

```diff
@@ -0,0 +1,260 @@
+#include <mg_generate.hpp>
+#include <mg_utils.hpp>
+
+#include "algorithm/label_propagation.hpp"
+
+constexpr char const *kWSelfloop = "w_selfloop";
+constexpr char const *kSimilarityThreshold = "similarity_threshold";
+constexpr char const *kExponent = "exponent";
+constexpr char const *kMinValue = "min_value";
+
+constexpr char const *kMaxIterations = "max_iterations";
+constexpr char const *kMaxUpdates = "max_updates";
+
+constexpr char const *kFieldNode = "node";
+constexpr char const *kFieldCommunity = "community";
+
+constexpr char const *kCreatedVertices = "createdVertices";
+constexpr char const *kCreatedEdges = "createdEdges";
+constexpr char const *kUpdatedVertices = "updatedVertices";
+constexpr char const *kUpdatedEdges = "updatedEdges";
+constexpr char const *kDeletedVertices = "deletedVertices";
+constexpr char const *kDeletedEdges = "deletedEdges";
+
+auto empty_graph = mg_generate::BuildGraph(0, {});
```

Comment:
I’ve defined `algorithm` as a global variable so that it can persist between function calls.
Simply declaring `LabelRankT::LabelRankT algorithm;` doesn’t work because the class lacks a default constructor, and in fact it can’t have one since the constructor takes a reference to the graph. To get around this, I initially define `algorithm` with a dummy empty graph.

Definitely open for a more elegant solution, tho!

Reply:
This is then a reasonable workaround, however, you might put it within algorithm implementation and not in the module, so the same structure can be used for unit-testing